WordPress Coding Standards fix for .scss files

ovidiul opened the pull request:
<!-- Start by describing the changes made in this Pull Request, and the reason for such changes. -->
This PR will add a fix for #3739 by adjusting WordPress Coding standards of `scss` files so values within parentheses don't have spaces padded.

<!-- Reference any related issues or PRs here -->
Fixes #3739 

<!-- Don't forget to update the title with something descriptive. -->
<!-- If your pull request implements a feature flag, make sure you update [this doc](../docs/blocks/features-and-blocks-behind-a-flag.md) -->

#### Accessibility

<!-- If you've changed or added any interactions, check off the appropriate items below. You can delete any that don't apply. Use this space to elaborate on anything if needed. -->

- [ ] I've tested using only a keyboard (no mouse)
- [ ] I've tested using a screen reader
- [ ] All animations respect [`prefers-reduced-motion`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion)
- [ ] All text has [at least a 4.5 color contrast with its background](https://webaim.org/resources/contrastchecker/)
- [ ] I've updated [this doc](../docs/blocks/feature-flags-and-experimental-interfaces.md) for any feature flags or experimental interfaces implemented in this pull request.

### Screenshots

<!-- If your change has a visual component, add a screenshot here. A "before" screenshot would also be helpful. -->

### Testing

#### Automated Tests
* [ ] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [ ] E2E tests
### Manual Testing

How to test the changes in this Pull Request:

1. no design changes here, just a coding standard fix, scss values and functionality remains the same
2. grep search for values with padded space within parenthesis will only show scss related logic which might not be related to the CSS coding standards
```
# grep -n --include \*.scss -R '( .* )' assets      

                                            
assets/css/abstracts/_breakpoints.scss:15:		@if type-of( $size ) == string {
assets/css/abstracts/_breakpoints.scss:52:				@warn "ERROR in breakpoint( #{ $size } ) : You can only use these sizes[ #{$sizes} ] using the following syntax [ <#{ nth( $breakpoints, 1 ) } >#{ nth( $breakpoints, 1 ) } #{ nth( $breakpoints, 1 ) }-#{ nth( $breakpoints, 2 ) } ]";
assets/css/abstracts/_breakpoints.scss:60:			@error "ERROR in breakpoint( #{ $size } ) : Please wrap the breakpoint $size in parenthesis. You can use these sizes[ #{$sizes} ] using the following syntax [ <#{ nth( $breakpoints, 1 ) } >#{ nth( $breakpoints, 1 ) } #{ nth( $breakpoints, 1 ) }-#{ nth( $breakpoints, 2 ) } ]";
```

<!-- If you can, add the appropriate labels -->

### Performance Impact

<!-- Please document any known performance impact (positive or negative) here. If negative, provide some rationale for why this is an okay tradeoff or how this will be addressed. -->

### Changelog

> fixes `scss` Wordpress coding standards so padded spaces are removed for values within parentheses

ovidiul committed:
adding timerange limit

ovidiul committed:
#3739 fixing scss coding standard and removing spaces from parantheses values

senadir commented:
Hey there @ovidiul  👋🏼 
The title of this PR and the description seems to be two separate things. The diff suggests both.
So to me, this PR does two unrelated things at the same time, can you break them or justify why do you need to have them both?

ovidiul commented:
> Hey there @ovidiul 👋🏼 The title of this PR and the description seems to be two separate things. The diff suggests both. So to me, this PR does two unrelated things at the same time, can you break them or justify why do you need to have them both?

My bad, seems I've fixed up the title, I've updated it now.

I Will fix my commit as well.

senadir commented:
The PR still introduces a timerange in `assets/js/blocks/product-best-sellers/block.js`, `assets/js/blocks/product-best-sellers/index.js`, and `src/BlockTypes/ProductBestSellers.php`, I can't see how that would be part of the PR?

ovidiul committed:
removing unrelated changes

ovidiul commented:
> The PR still introduces a timerange in `assets/js/blocks/product-best-sellers/block.js`, `assets/js/blocks/product-best-sellers/index.js`, and `src/BlockTypes/ProductBestSellers.php`, I can't see how that would be part of the PR?

Commit fixed as well, I was just about to reply on that. Thanks!

opr reviewed (changes requested):
Hey @ovidiul thanks for your contribution, so far this looks good to me but I found a few other instances of parentheses with spaces. I've included a patch file below showing the differences.

```patch
diff --git a/assets/css/abstracts/_breakpoints.scss b/assets/css/abstracts/_breakpoints.scss
index a4a373e6..d191574a 100644
--- a/assets/css/abstracts/_breakpoints.scss
+++ b/assets/css/abstracts/_breakpoints.scss
@@ -12,7 +12,7 @@ $breakpoints: 480px, 600px, 782px, 960px, 1280px, 1440px;
 // https://github.com/WordPress/gutenberg/blob/master/packages/base-styles/_mixins.scss
 @mixin breakpoint($sizes...) {
 	@each $size in $sizes {
-		@if type-of( $size ) == string {
+		@if type-of($size) == string {
 			$approved-value: 0;
 			@each $breakpoint in $breakpoints {
 				$and-larger: ">" + $breakpoint;
@@ -49,7 +49,7 @@ $breakpoints: 480px, 600px, 782px, 960px, 1280px, 1440px;
 				@each $breakpoint in $breakpoints {
 					$sizes: $sizes + " " + $breakpoint;
 				}
-				@warn "ERROR in breakpoint( #{ $size } ) : You can only use these sizes[ #{$sizes} ] using the following syntax [ <#{ nth( $breakpoints, 1 ) } >#{ nth( $breakpoints, 1 ) } #{ nth( $breakpoints, 1 ) }-#{ nth( $breakpoints, 2 ) } ]";
+				@warn "ERROR in breakpoint(#{ $size }) : You can only use these sizes[ #{$sizes} ] using the following syntax [ <#{ nth($breakpoints, 1) } >#{ nth($breakpoints, 1) } #{ nth($breakpoints, 1) }-#{ nth($breakpoints, 2) } ]";
 			}
 		}
 		@else {
@@ -57,7 +57,7 @@ $breakpoints: 480px, 600px, 782px, 960px, 1280px, 1440px;
 			@each $breakpoint in $breakpoints {
 				$sizes: $sizes + " " + $breakpoint;
 			}
-			@error "ERROR in breakpoint( #{ $size } ) : Please wrap the breakpoint $size in parenthesis. You can use these sizes[ #{$sizes} ] using the following syntax [ <#{ nth( $breakpoints, 1 ) } >#{ nth( $breakpoints, 1 ) } #{ nth( $breakpoints, 1 ) }-#{ nth( $breakpoints, 2 ) } ]";
+			@error "ERROR in breakpoint(#{ $size }) : Please wrap the breakpoint $size in parenthesis. You can use these sizes[ #{$sizes} ] using the following syntax [ <#{ nth($breakpoints, 1) } >#{ nth($breakpoints, 1) } #{ nth($breakpoints, 1) }-#{ nth($breakpoints, 2) } ]";
 		}
 	}
 }
diff --git a/assets/js/blocks/cart-checkout/payment-methods/style.scss b/assets/js/blocks/cart-checkout/payment-methods/style.scss
index 488121e6..1a24d012 100644
--- a/assets/js/blocks/cart-checkout/payment-methods/style.scss
+++ b/assets/js/blocks/cart-checkout/payment-methods/style.scss
@@ -31,7 +31,7 @@
 		// Notice the min width ems value is smaller than flex-basis. That's because
 		// by default we want it to have the same width as `expiry-element`, but
 		// if available space is scarce, `cvc-element` should get smaller faster.
-		min-width: unquote("min( 5em, calc(16% - #{$gap-small}))");
+		min-width: unquote("min(5em, calc(16% - #{$gap-small}))");
 	}
 
 	.wc-block-gateway-input {
diff --git a/packages/checkout/panel/style.scss b/packages/checkout/panel/style.scss
index 2fc5692f..989e5f08 100644
--- a/packages/checkout/panel/style.scss
+++ b/packages/checkout/panel/style.scss
@@ -1,5 +1,5 @@
 .wc-block-components-panel.has-border {
-	@include with-translucent-border( 1px 0 );
+	@include with-translucent-border(1px 0);
 
 	+ .wc-block-components-panel.has-border::after {
 		border-top-width: 0;
@@ -7,7 +7,7 @@
 }
 
 .wc-block-components-panel.has-border.no-top-border {
-	@include with-translucent-border( 1px 0 );
+	@include with-translucent-border(1px 0);
 	&::after {
 		border-top-width: 0;
 	}
diff --git a/packages/checkout/wrapper/style.scss b/packages/checkout/wrapper/style.scss
index cde84456..32bbf8f9 100644
--- a/packages/checkout/wrapper/style.scss
+++ b/packages/checkout/wrapper/style.scss
@@ -1,5 +1,5 @@
 .wc-block-components-totals-wrapper {
-	@include with-translucent-border( 1px 0 0 );
+	@include with-translucent-border(1px 0 0);
 	padding: $gap 0;
 
 	&.has-bottom-border {
@@ -12,7 +12,7 @@
 		padding: 0;
 
 		> * > * {
-			@include with-translucent-border( 0 0 1px );
+			@include with-translucent-border(0 0 1px);
 			padding: $gap 0;
 
 			&:last-child::after {
@@ -25,7 +25,7 @@
 .wc-block-components-discounts-meta {
 	.wc-block-components-totals-wrapper {
 		&:first-child {
-			@include with-translucent-border( 1px 0 0 );
+			@include with-translucent-border(1px 0 0);
 		}
 	}
 }
```

If you make these changes then I can approve. Thanks again!

ovidiul committed:
fixing further instances for parantheses

ovidiul commented:
@opr patch has been applied. Thank you as well!

opr reviewed (approved):
Thanks for making the changes, and thanks for your contribution to the project!

I will merge the changes now.

FYI The `npm run lint:js` job fails in CI because this PR comes from a fork and I think it can't upload the artefact, but if you see here the actual lint task runs OK.

https://github.com/woocommerce/woocommerce-gutenberg-products-block/runs/3859106771?check_suite_focus=true#step:5:11

✅ from me!
